kvbc/test/sparse_merkle_storage/kv_blockchain_db_editor_test.cpp

```diff
@@ -0,0 +1,567 @@
+  ASSERT_EQ(
+      EXIT_SUCCESS,
+      run(CommandLineArguments{{kTestName, rocksDbPath(main_path_db_id_), "getLastStateTransferBlockID"}}, out_, err_));
+  ASSERT_TRUE(err_.str().empty());
+  ASSERT_EQ("{\n  \"lastStateTransferBlockID\": \"n/a\"\n}\n", out_.str());
+}
+
+TEST_F(DbEditorTests, get_block_info) {
+  ASSERT_EQ(EXIT_SUCCESS,
+            run(CommandLineArguments{{kTestName, rocksDbPath(main_path_db_id_), "getBlockInfo", "5"}}, out_, err_));
+  ASSERT_TRUE(err_.str().empty());
+  ASSERT_THAT(out_.str(), HasSubstr("  \"parentBlockDigest\": \"0x"));
+  ASSERT_THAT(out_.str(), HasSubstr("  \"keyValueCount\": \"" + std::to_string(num_keys_ * num_categories_) + '\"'));
+  ASSERT_THAT(out_.str(), HasSubstr("  \"categoriesCount\": \"" + std::to_string(num_categories_) + '\"'));
+}
+
+TEST_F(DbEditorTests, get_block_info_missing_block_id) {
+  ASSERT_EQ(EXIT_FAILURE,
+            run(CommandLineArguments{{kTestName, rocksDbPath(main_path_db_id_), "getBlockInfo"}}, out_, err_));
+  ASSERT_TRUE(out_.str().empty());
+  ASSERT_THAT(err_.str(), StartsWith("Failed to execute command [getBlockInfo], reason: Missing BLOCK-ID argument"));
+}
+
+TEST_F(DbEditorTests, get_block_key_values) {
+  ASSERT_EQ(
+      EXIT_SUCCESS,
+      run(CommandLineArguments{{kTestName, rocksDbPath(main_path_db_id_), "getBlockKeyValues", "5"}}, out_, err_));
+  ASSERT_TRUE(err_.str().empty());
+  ASSERT_THAT(out_.str(), StartsWith("{\n\"" + kCategoryMerkle + "\": {\n  \"0x"));
+  ASSERT_THAT(out_.str(), HasSubstr("\"0x0000003c\": \"0x00000078\""));
+  ASSERT_THAT(out_.str(), HasSubstr("\"0x00000028\": \"0x00000050\""));
+  ASSERT_THAT(out_.str(), HasSubstr("\"0x00000032\": \"0x00000064\""));
+  ASSERT_THAT(out_.str(), EndsWith("\n}\n"));
+}
+
+TEST_F(DbEditorTests, get_empty_block_key_values) {
+  ASSERT_EQ(EXIT_SUCCESS,
+            run(
+                CommandLineArguments{
+                    {kTestName, rocksDbPath(main_path_db_id_), "getBlockKeyValues", std::to_string(empty_block_id_)}},
+                out_,
+                err_));
+  ASSERT_TRUE(err_.str().empty());
+  ASSERT_EQ("{\n\"" + kCategoryMerkle + "\": {\n},\n\"" + kCategoryVersioned + "\": {\n}\n}\n", out_.str());
+}
+
+TEST_F(DbEditorTests, get_block_key_values_missing_block_id) {
+  ASSERT_EQ(EXIT_FAILURE,
+            run(CommandLineArguments{{kTestName, rocksDbPath(main_path_db_id_), "getBlockKeyValues"}}, out_, err_));
+  ASSERT_TRUE(out_.str().empty());
+  ASSERT_THAT(err_.str(),
+              StartsWith("Failed to execute command [getBlockKeyValues], reason: Missing BLOCK-ID argument"));
+}
+
+TEST_F(DbEditorTests, get_value_latest) {
+  ASSERT_EQ(
+      EXIT_SUCCESS,
+      run(CommandLineArguments{{kTestName, rocksDbPath(main_path_db_id_), "getValue", kCategoryMerkle, "0x00000028"}},
+          out_,
+          err_));
+  ASSERT_TRUE(err_.str().empty());
+  ASSERT_EQ("{\n  \"value\": \"0x00000050\"\n}\n", out_.str());
+}
+
+TEST_F(DbEditorTests, get_value_with_block_version) {
```

Comment:
Another test that can be added is to call `getValue` with a non-existent block version. Maybe a version bigger than the latest block ID. I guess `getValue` with version 0 below handles it, but might still be a good test to add.